contracts/V1/AStateMachine.sol

```diff
@@ -60,33 +81,43 @@ abstract contract AStateMachine {
 
     function slashParticipant(
         address adr
-    ) external _nonReentrant returns (bool, ProcessExit memory) {
-        return _slashParticipant(adr);
+    ) external _nonReentrant returns (bool, ExitChannel memory) {
+        (bool success, ExitChannel memory exitChannel) = _slashParticipant(adr);
+        if (success) {
+            _addExitChannel(exitChannel);
+        }
+        return (success, exitChannel);
     }
 
     function removeParticipant(
         address adr
-    ) external virtual _nonReentrant returns (bool, ProcessExit memory) {
+    ) external virtual _nonReentrant returns (bool, ExitChannel memory) {
         return _removeParticipant(adr);
     }
 
     function stateTransition(
         Transaction calldata transaction
-    ) external _nonReentrant returns (bool) {
+    )
+        external
+        _nonReentrant
+        returns (bool success, ExitChannel[] memory exitChannels)
+    {
         _tx = transaction;
-        (bool success, bytes memory result) = address(this).call(
+        _clearExitChannels();
+        (bool callSuccess, bytes memory result) = address(this).call(
             transaction.body.data
         );
         // emit TxExecutedA(success, getState());
-        if (!success) {
+        if (!callSuccess) {
             if (result.length == 0)
                 revert("AStateMachine - Call failed - result lenght 0");
             assembly {
                 let returndata_size := mload(result)
                 revert(add(32, result), returndata_size)
             }
         }
-        return success;
+
+        return (callSuccess, _exitChannels);
```

Comment:
We can keep this function as just 'return bool' since both the 'state' and 'exitChannels' can be fetched 'getState' 'getExitChannels' if needed by the calling function/context if needed